### PR TITLE
Add checks on sk_TYPE_push() returned status in ssl module.

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
PR related to ssl folder only ...
More to come in separated PRs.

### CODE REVIEW

File: ssl/d1_srtp.c [hunk 1]
```diff
@@ -81,16 +81,18 @@ static int ssl_ctx_make_profiles(const char *profiles_string,
...
                 SSLerr(SSL_F_SSL_CTX_MAKE_PROFILES,
                        SSL_R_BAD_SRTP_PROTECTION_PROFILE_LIST);
-                sk_SRTP_PROTECTION_PROFILE_free(profiles);
-                return 1;
+                goto err;
             }
 
-            sk_SRTP_PROTECTION_PROFILE_push(profiles, p);
+            if (!sk_SRTP_PROTECTION_PROFILE_push(profiles, p)){
+                SSLerr(SSL_F_SSL_CTX_MAKE_PROFILES,
```
Comment (richsalz) [inline ssl/d1_srtp.c]: should me err_r_malloc, right?

File: ssl/ssl_cert.c [hunk 2]
```diff
@@ -657,16 +662,20 @@ int SSL_add_file_cert_subjects_to_stack(STACK_OF(X509_NAME) *stack,
         if (xn == NULL)
             goto err;
         if (sk_X509_NAME_find(stack, xn) >= 0)
+            /* Duplicate. */
             X509_NAME_free(xn);
-        else
-            sk_X509_NAME_push(stack, xn);
+        else
```
Comment (richsalz) [inline ssl/ssl_cert.c]: should be "else if" on one line.

File: ssl/ssl_cert.c [hunk 3]
```diff
@@ -657,16 +662,20 @@ int SSL_add_file_cert_subjects_to_stack(STACK_OF(X509_NAME) *stack,
         if (xn == NULL)
             goto err;
         if (sk_X509_NAME_find(stack, xn) >= 0)
+            /* Duplicate. */
```
Comment (richsalz) [inline ssl/ssl_cert.c]: since the else has curly braces, style says this needs  curly braces.

File: ssl/d1_srtp.c (hunk 1 above)
Comment (FdaSilvaYY) [inline ssl/d1_srtp.c]: Fixed, I'm not sure it's useful...

**Comment by richsalz** [comment]
malloc failures are not necessarily the best use of our time.  sigh.  
+1

File: crypto/engine/eng_dyn.c [hunk 4]
```diff
@@ -353,7 +353,10 @@ static int dynamic_ctrl(ENGINE *e, int cmd, long i, void *p, void (*f) (void))
                 ENGINEerr(ENGINE_F_DYNAMIC_CTRL, ERR_R_MALLOC_FAILURE);
                 return 0;
             }
-            sk_OPENSSL_STRING_insert(ctx->dirs, tmp_str, -1);
+            if (!sk_OPENSSL_STRING_insert(ctx->dirs, tmp_str, -1)){
```
Comment (kroeckx) [inline crypto/engine/eng_dyn.c]: Can you change the !tmp_str to tmp_str == NULL just above this?

Comment (kroeckx) [inline crypto/engine/eng_dyn.c]: I wonder why it's not using push instead of insert with -1

File: ssl/d1_srtp.c [hunk 5]
```diff
@@ -81,16 +81,18 @@ static int ssl_ctx_make_profiles(const char *profiles_string,
             if (sk_SRTP_PROTECTION_PROFILE_find(profiles, p) >= 0) {
                 SSLerr(SSL_F_SSL_CTX_MAKE_PROFILES,
                        SSL_R_BAD_SRTP_PROTECTION_PROFILE_LIST);
-                sk_SRTP_PROTECTION_PROFILE_free(profiles);
-                return 1;
+                goto err;
             }
 
-            sk_SRTP_PROTECTION_PROFILE_push(profiles, p);
+            if (!sk_SRTP_PROTECTION_PROFILE_push(profiles, p)){
```
Comment (kroeckx) [inline ssl/d1_srtp.c]: Add a space before the {

File: ssl/d1_srtp.c (hunk 1 above)
Comment (kroeckx) [inline ssl/d1_srtp.c]: I don't see it fixed?

File: ssl/ssl_cert.c [hunk 6]
```diff
@@ -656,17 +661,20 @@ int SSL_add_file_cert_subjects_to_stack(STACK_OF(X509_NAME) *stack,
         xn = X509_NAME_dup(xn);
         if (xn == NULL)
             goto err;
-        if (sk_X509_NAME_find(stack, xn) >= 0)
+        if (sk_X509_NAME_find(stack, xn) >= 0) {
+            /* Duplicate. */
             X509_NAME_free(xn);
-        else
-            sk_X509_NAME_push(stack, xn);
+        } else if(!sk_X509_NAME_push(stack, xn)) {
```
Comment (kroeckx) [inline ssl/ssl_cert.c]: space after if

File: crypto/engine/eng_dyn.c [hunk 7]
```diff
@@ -349,11 +349,14 @@ static int dynamic_ctrl(ENGINE *e, int cmd, long i, void *p, void (*f) (void))
         }
         {
             char *tmp_str = OPENSSL_strdup(p);
-            if (!tmp_str) {
+            if (tmp_str == NULL) {
                 ENGINEerr(ENGINE_F_DYNAMIC_CTRL, ERR_R_MALLOC_FAILURE);
                 return 0;
             }
-            sk_OPENSSL_STRING_insert(ctx->dirs, tmp_str, -1);
+            if (!sk_OPENSSL_STRING_push(ctx->dirs, tmp_str)){
```
Comment (mattcaswell) [inline crypto/engine/eng_dyn.c]: Add space before final open brace

File: crypto/engine/eng_dyn.c [hunk 8]
```diff
@@ -349,11 +349,14 @@ static int dynamic_ctrl(ENGINE *e, int cmd, long i, void *p, void (*f) (void))
...
+                OPENSSL_free(tmp_str);
```
Comment (mattcaswell) [inline crypto/engine/eng_dyn.c]: Shouldn't there be an ENGINEerr call here?

File: ssl/ssl_cert.c [hunk 9]
```diff
@@ -474,6 +474,7 @@ STACK_OF(X509_NAME) *SSL_dup_CA_list(STACK_OF(X509_NAME) *sk)
         name = X509_NAME_dup(sk_X509_NAME_value(sk, i));
```
Comment (mattcaswell) [inline ssl/ssl_cert.c]: While we're adding these checks, we probably should also check `ret` for NULL

Comment (FdaSilvaYY) [inline ssl/ssl_cert.c]: done !

**Comment by mattcaswell** [comment]
+1
@richsalz reconfirm?

**Comment by richsalz** [comment]
+1
you going to merge or should i?

**Comment by mattcaswell** [comment]
Merged. Thanks.

**Pull request closed without merging**